src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1744,4 +1760,113 @@ void RaftReplDev::report_blk_metrics_if_needed(repl_req_ptr_t rreq) {
     }
 }
 
+void RaftReplDev::pause_statemachine() {
+    if (!raft_server()->is_state_machine_execution_paused()) {
+        raft_server()->pause_state_machine_exeuction();
+        while (!raft_server()->wait_for_state_machine_pause(100)) {
+            RD_LOGD(NO_TRACE_ID, "wait for statemachine pause!");
+        }
+    }
+}
+
+void RaftReplDev::resume_statemachine() {
+    if (raft_server()->is_state_machine_execution_paused()) {
+        raft_server()->resume_state_machine_execution();
+        RD_LOGD(NO_TRACE_ID, "statemachine is resumed!");
+    }
+}
+
+void RaftReplDev::enter_emergency() {
```

Comment:
we can rename it to something like `drain_outstanding_rreqs`, since it is some fact API homestore can provide to consumer to do stuff.